adsdata/nonbib.py

```diff
@@ -335,6 +343,30 @@ def count_lines(self, file):
             on {0}.canonical.bibcode = {1}.canonical.bibcode \
             where {1}.canonical.bibcode IS NULL;'
 
+    # resolver
+    include_new_resolver_bibcodes_sql = \
+        'insert into {0}.ChangedRowsM (bibcode) \
+            select distinct on (bibcode) bibcode from \
+               (select {0}.datalinks.bibcode from {0}.datalinks left join {1}.datalinks \
+                on {0}.datalinks.bibcode = {1}.datalinks.bibcode \
+                and {0}.datalinks.link_type = {1}.datalinks.link_type \
+                and {0}.datalinks.link_sub_type = {1}.datalinks.link_sub_type \
+                where {0}.datalinks.url != {1}.datalinks.url \
+                or {0}.datalinks.title != {1}.datalinks.title \
+                or {0}.datalinks.item_count != {1}.datalinks.item_count \
+                or {1}.datalinks.bibcode IS NULL) as datalinks;'
+
+    populate_new_resolver_bibcodes_sql = \
+        'insert into {0}.newbibcodes (bibcode) \
+            select distinct on (bibcode) bibcode from \
+               (select {0}.datalinks.bibcode from {0}.datalinks left join {1}.datalinks \
+                on {0}.datalinks.bibcode = {1}.datalinks.bibcode \
+                and {0}.datalinks.link_type = {1}.datalinks.link_type \
+                and {0}.datalinks.link_sub_type = {1}.datalinks.link_sub_type \
+                where {0}.datalinks.url != {1}.datalinks.url \
+                or {0}.datalinks.title != {1}.datalinks.title \
+                or {0}.datalinks.item_count != {1}.datalinks.item_count \
+                or {1}.datalinks.bibcode IS NULL) as datalinks;'
```

Comment:
I think here I made a mistake when I answered one of your questions via slack, I'm sorry! I think I made things more confusing. In this part of the code we only want to get the new bibcodes, the conditions need to be only:
```
... where {1}.datalinks.bibcode IS NULL) as datalinks;
```
If we include all the other conditions, then we will select also bibcodes that are not new but have been updated.

Reply:
so that was my point that there was no difference between new and populate. however, looked at the other two sql statements above that are exactly the same. that is why I went this way. I agree that it should be different. give me a go ahead one more time and I shall change it.